k2/python/csrc/torch/v2/ragged_arc.cu

```diff
@@ -58,6 +64,151 @@ RaggedArc::RaggedArc(
   // TODO: we also need to pass the name of extra_labels and ragged_labels.
 }
 
+RaggedArc::RaggedArc(const RaggedArc &src, const Ragged<Arc> &arcs,
+                     torch::Tensor arc_map) {
+  fsa = arcs;
+  for (auto iter = src.tensor_attrs.begin(); iter != src.tensor_attrs.end();
+      ++iter) {
+    float filler = GetFiller(iter->first);
+    auto value = IndexSelectFunction::apply(iter->second, arc_map, filler);
+    SetAttr(iter->first, value);
+  }
+
+  for (auto iter = src.ragged_tensor_attrs.begin();
+      iter != src.ragged_tensor_attrs.end(); ++iter) {
+    // Only integer types ragged attributes are supported now
+    K2_CHECK_EQ(iter->second.any.GetDtype(), kInt32Dtype);
+    auto value = const_cast<RaggedAny &>(iter->second).Index(arc_map, 0, false);
+    SetAttr(iter->first, value.first);
+  }
+
+  for (auto iter = src.other_attrs.begin(); iter != src.other_attrs.end();
+      ++iter) {
+    SetAttr(iter->first, iter->second);
+  }
+  // populate scores
+  Scores();
+  IndexSelectScoresFunction::apply(*this, src.Scores(), arc_map);
+}
+
+RaggedArc::RaggedArc(const RaggedArc &src, const Ragged<Arc> &arcs,
+                     RaggedAny &arc_map, bool remove_filler /*= true*/) {
+  fsa = arcs;
+  for (auto iter = src.tensor_attrs.begin(); iter != src.tensor_attrs.end();
+      ++iter) {
+    if (remove_filler && iter->second.scalar_type() == torch::kInt32) {
+      float filler = GetFiller(iter->first);
+      if (filler != -1) {
+        torch::Tensor value = iter->second.clone();
+        value = torch::where(
+            torch::logical_or(
+              torch::ne(const_cast<RaggedArc &>(src).Labels(), -1),
```

Comment:
We can pass `RaggedArc&` instead of `const RaggedArc` so that you don't need to use `const_cast` here.